changelogs/unreleased/3076-fix-memory-leak-introduced-by-using-rlu-cache.yml

```diff
@@ -0,0 +1,6 @@
+description: Fix memory leak caused by lru-cache keeping strong references to cached items
+change-type: patch
+destination-branches: [master, iso5]
```

Comment:
This bug also exists on iso4. Are you planning to create a separate PR for ISO4? That could be useful since a lot of changes were done to the module.py file between ISO4 and ISO5 and doing everything in one PR might cause a big merge conflict. I am not sure what is best.

Reply:
I will create a separate PR for ISO4 then.